docs/async/async-csharp-vb.rst

```diff
@@ -0,0 +1,202 @@
+Async Programming in C#/VB
+==========================
+By `Phillip Carter`_
+
+From a developer's perspective, the C#/VB async model is some nice syntax that hides complexity and allows you to focus on writing methods without worrying about things like thread pools and callbacks.
+
+The core of this are the ``async`` and ``await`` keywords (``Async`` and ``Await`` in VB.NET), which you sprinkle over otherwise normal methods. 
+
+A few important things to know before continuing:
+
+* ``Task<T>`` and ``Task`` are used to represent the return types for async code.  These are abstractions that can be thought of as "something that needs to get done and could be long-running".  `More on Task and Task<T>`_.
+* Marking a method as ``async`` tells the compiler that you're expecting to ``await`` some blocking task.
+* ``await`` can only be used inside an async method.
+* When the ``await`` keyword is applied, it suspends the async method and returns control back to its caller until the awaited task is complete.  This is what makes it responsive.
+* Unless an async method has an ``await`` inside its body, it will run synchronously!
+* Return types for async methods should always be either ``Task<T>`` or ``Task``, with the exception of event handlers (these are ``void``).  More on this later.
+
+Simple Async Example (C#)
+-------------------------
+
+The following example shows somewhat of a "real life" scenario: getting directions to a user for a ridesharing app.
+
+Client app snippet:
+
+.. code-block:: c#
+
+	private async void GetDirectionsForPickup_Pressed(object sender, CustomUserDataEventArgs e)
+	{
+	    var user = e.Source as BasicUserData;
```

Comment:
I think that this example is too complicated for a simple example. I propose a simple HttpClient call and deserialization of JSON data, and may updating a label.